lib/rspec/core/profiler.rb

```diff
@@ -0,0 +1,29 @@
+module RSpec
+  module Core
+    # @private
+    class Profiler
+      NOTIFICATIONS = [:example_group_started, :example_group_finished, :example_started]
+
+      def initialize
+        @example_groups = Hash.new { |h, k| h[k] = Hash.new(0) }
```

Comment:
This line confuses me a bit, because it suggests that all the nested hashes should have incrementing integer values...but `:start`, `:description` and `:total_time` are not integers.  WDYT of this instead?

```` ruby
@example_groups = Hash.new { |h, k| h[k] = { :count => 0 } }
```

That provides the count default of `0` (so you can simply increment it below w/o worrying about if it's the first time or not) without confusingly suggesting all the entries in the nested hash are integers.